op-program/client/l1/oracle.go

```diff
@@ -86,3 +92,26 @@ func (p *PreimageOracle) ReceiptsByBlockHash(blockHash common.Hash) (eth.BlockIn
 
 	return info, receipts
 }
+
+func (p *PreimageOracle) GetBlob(ref eth.L1BlockRef, blobHash eth.IndexedBlobHash) *eth.Blob {
+	// Send a hint for the blob commitment & blob field elements.
+	blobReqMeta := make([]byte, 16)
+	binary.BigEndian.PutUint64(blobReqMeta[0:8], blobHash.Index)
+	binary.BigEndian.PutUint64(blobReqMeta[8:16], ref.Time)
+	p.hint.Hint(BlobHint(append(blobHash.Hash[:], blobReqMeta...)))
+
+	commitment := p.oracle.Get(preimage.Sha256Key(blobHash.Hash))
+
+	// Reconstruct the full blob from the 4096 field elements.
+	blob := eth.Blob{}
+	fieldElemKey := make([]byte, 80)
+	copy(fieldElemKey[:48], commitment)
+	for i := 0; i < params.BlobTxFieldElementsPerBlob; i++ {
+		binary.BigEndian.PutUint64(fieldElemKey[72:], uint64(i))
+		fieldElement := p.oracle.Get(preimage.BlobKey(crypto.Keccak256(fieldElemKey)))
+
+		copy(blob[i<<5:(i+1)<<5], fieldElement[:])
+	}
+
```

Comment:
It would be nice to add a sanity-check that the blob indeed matches the KZG commitment (and a comment that it's a sanity check, the construction already forces the blob to be correct).

You can call `blob.ComputeKZGCommitment()` to compute it, and then compare it against the `commitment`, and panic if it doesn't match.

Reply:
Can also unit test this to verify this panics by returning the invalid commitment from a stub oracle which will not match the computed `blobCommitment`.